### DISCUSSION TITLE
Sonoff T0UK1C with a T1 V1 board

### DISCUSSION THREAD

**Discussion by linker3000** [arendst/Tasmota#17934] [Show and tell]
Just in case it helps someone, I purchased a couple of second hand Sonoff T0UK1C touch light switches off the 'bay and checked out the Tasmota flashing instructions pages, but my switches' main logic boards did not match any of those shown in the info, or on the general www, for either the T0 or T1 variants and, somewhat confusingly, the 'T0' unit's boards were labelled T1 V1.0.

Having traced the PCB track for GPIO0, which needs to be grounded on power-up to set the ESP8285 into flash mode, I found it on a via close to the chip (see pic), so all that's needed is to scrape off the solder mask and ground the via - I used one of my multimeters (switched off, not that it would have mattered) with fine probes, set to current (low resistance) to link the via to GND.

Having done that, it took a few seconds with ESP-Flasher to change the firmware to Tasmota and both switches are working well.

![IMG_20230211_164114144](https://user-images.githubusercontent.com/19429471/218278957-0973fc4e-c81c-42ee-a526-c18d18472587.jpg)

**Comment by sfromis** [comment]
You need no scraping, you can just touch gpio0 directly on the ESP8285 chip (bottom edge, 2nd from the right). Not hard to do, as it fortunately is perfectly safe to also ground the neighbor to the right (gpio4) at the same time.

**Reply by linker3000** [reply]
Thanks for the feedback. I am always wary of directly connecting to the pins, expecially because 14 (GPIO2) is an active output during the flashing process. The via is also convenient because, as a hole, the tip of a fine multimeter probe fits nicely and is less likely to slip in any direction.

**Reply by sfromis** [reply]
The tip also rests nicely between two pins of the ESP8285. And if you happen to hit gpio2, no big deal, at max you may have to restart flashing, as boot fails if you pull it low.

The problem with scraping is that different people may have different ideas about what being gentle and careful means, leading to risk of cutting traces, tricky to repair. Good reasons to be wary about this. No real damage risk by touching the pins of the ESP8285, as long as you stay within reasonable voltages.
